### PR TITLE
[az fleet]  Creation of dedicated "fleet-preview" extension.

### PR DESCRIPTION
**Pull request by Ealianis**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### Related command
This PR clones the existing Azure Kubernetes Fleet extension and modifies it into a dedicated preview extension. The command signatures are the same, e.g.,  `az fleet`, therefore a user can only have one Fleet extension (preview or non-preview) installed at a time. 

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally? (`pip install wheel==0.30.0` required)

For new extensions:

- [x] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
Azure Kubernetes Fleet

File: src/fleet-preview/setup.py
Comment (jim-minter) [inline src/fleet-preview/setup.py:57]:
```suggestion
    package_data={'azext_fleet_preview': ['azext_metadata.json']},
```

**Review by jim-minter — CHANGES_REQUESTED** [review]
Please hold this PR until we (Fleet) are sure we want to merge it

**Pull request closed without merging**
